openedx/course-discovery
# ENT-657: Change UTM Source value in API results returned by discovery API

Hi,

This PR contains changes for [ENT-657](https://openedx.atlassian.net/browse/ENT-657).

**Note:**
Currently I am working on the tests for the changes, I will update the PR once tests are ready.

**Jira Ticket:**   [ENT-652](https://openedx.atlassian.net/browse/ENT-652) and [ENT-657](https://openedx.atlassian.net/browse/ENT-657)

**Description:**
The `MinimalCourseRunSerialize`r includes the `marketing_url` and appends the UTM parameters. We will need to modify the `get_marketing_url_for_user` function which appends the UTM parameters to the `marketing_url`.
The information that the partner enters on the API access request form is stored in the LMS database using the `ApiAccessRequest` model. We will need to add an API view in the `api_admin` Django app which will return the `ApiAccessRequest` records for the current user. The `api_admin` data model allows for a single user to have many `ApiAccessRequest` records. The discovery service will need to make call the `api_admin` API endpoint, pick the first `ApiAccessRequest` record returned, and use the `company_name` to populate the `utm_source` value on the marketing URL.

**Testing Instructions:**

**Note:** You can get the company name for some user by logging in the LMS with staff user and accessing https://business.sandbox.edx.org/api-admin/api/v1/api_access_request/?user__username=<username> just replace `<username>` with appropriate value.

1. Login to [discovery-business.sandbox.edx.org](https://discovery-business.sandbox.edx.org) using `staff` user.
2. Go to [Waffle Admin Page for `use_company_name_as_utm_source_value`](https://discovery-business.sandbox.edx.org/admin/waffle/switch/12/change/) and uncheck the `active` check button so that waffle switch is disabled.
3. Open discovery API's catalog details page (catalog must be associated with the current user) https://discovery-business.sandbox.edx.org/api/v1/catalogs/5/courses/ and make sure `utm_source` in `marketing_url` is set to the username of the logged in user.
4. Go back o [Waffle Admin Page for `use_company_name_as_utm_source_value`](https://discovery-business.sandbox.edx.org/admin/waffle/switch/12/change/) and check the `active` check button so that waffle switch is enabled.
5. Open discovery API's catalog details page (catalog must be associated with the current user) https://discovery-business.sandbox.edx.org/api/v1/catalogs/5/courses/ and make sure `utm_source` in `marketing_url` is set to the "\<username\>-\<company-name\>".
6. Create a new user and go to [LMS API Admin Page](https://business.sandbox.edx.org/api-admin/) and add a new `APIAccessRequest`. 
7. Login to a separate browser with `staff` user and got [APIAccessRequest Admin Page](https://business.sandbox.edx.org/admin/api_admin/apiaccessrequest/) and approve `APIAccessRequest` for the new user.
8. Now, repeat 2 through 5 points from above to verify discovery's API responses for the new user
